File: 프로그래머스/lv2/42584. 주식가격/주식가격.py
from collections import deque

def solution(prices):
    answer = []
    for i in range(len(prices)):
        cnt = 0
        low_price = prices[i]
        for j in range(i+1, len(prices)):
            cnt += 1
            if low_price <= prices[j]:
                continue
            else:
                break
        answer.append(cnt)
    return answer

File: 프로그래머스/lv2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Queue<Integer> queue = new LinkedList<>();
        
        for (int price : prices) {
            queue.add(price);
        }

        while (!queue.isEmpty()) {
            for (int i = 0; i < prices.length; i++) {
                int cnt = 0;
                int cur_val = queue.poll();
                for (int compare_val : queue) {
                    cnt++;
                    if (cur_val > compare_val) {
                        break;
                    }
                }
                answer[i] = cnt;
            }
        }

        return answer;
    }
}